### PR TITLE
refactor: remove useAsyncMemo

### LINKED ISSUE #1014: Improve frontend caching strategy
- [x] Profile performance before changes.
- [x] Using react-query improve frontend caching strategy.
- [ ] Profile performance after deployment to compare.

### PR DESCRIPTION
**Pull request by ncomerci**
This closes #1014

### CODE REVIEW

File: src/hooks/useProposals.ts [hunk 1]
```diff
@@ -11,74 +11,63 @@ export type UseProposalsFilter = Omit<GetProposalsFilter, 'subscribed' | 'limit'
...
-    filter.timeFrameKey,
-    filter.order,
-    filter.page,
-    filter.itemsPerPage,
-    filter.load,
-    filter.coauthor,
-    filter.snapshotIds,
-    filter.subtype,
-  ])
+      return Governance.get().getProposals({ ...params, limit, offset })
```
Comment (1emu) [inline src/hooks/useProposals.ts:64]: bby dont hurt me

File: src/hooks/useSurveyTopics.ts [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+    queryFn: async () => {
+      if (!proposalId) {
+        return null
+      }
       return Governance.get().getSurveyTopics(proposalId)
     },
-    [proposalId],
-    { callWithTruthyDeps: true }
-  )
+    staleTime: 3.6e6, // 1 hour
```
Comment (1emu) [inline src/hooks/useSurveyTopics.ts:14]: no more

File: src/hooks/useVotesMatch.ts [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
+    staleTime: 3.6e6, // 1 hour
+  })
+
+  const { data: otherAccountVotes, isLoading: otherAccountVotesLoading } = useQuery({
+    queryKey: [`otherAccountVotes#${otherAccount}`],
+    queryFn: async () => {
+      if (!otherAccount) {
+        return null
+      }
+      return Governance.get().getAddressVotes(otherAccount)
```
Comment (1emu) [inline src/hooks/useVotesMatch.ts:25]: both queryFn here may or may not be missing something

**Review by 1emu — APPROVED** [review]
image

File: src/hooks/useCoauthors.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { useQuery } from '@tanstack/react-query'
+
+import { Governance } from '../clients/Governance'
+import { CoauthorStatus } from '../entities/Coauthor/types'
+
+function useCoauthors(proposalId: string, status?: CoauthorStatus) {
+  const { data: coAuthors } = useQuery({
+    queryKey: [`coauthors-${proposalId}`],
+    queryFn: () => Governance.get().getCoAuthorsByProposal(proposalId, status),
```
Comment (andyesp) [inline src/hooks/useCoauthors.ts:9]: You can handle here if there is no proposalId instead of passing an empty string in some cases to avoid querying it.

Comment (andyesp) [reply src/hooks/useCoauthors.ts:9]: https://tanstack.com/query/v3/docs/react/guides/dependent-queries

File: src/hooks/useIsDebugAddress.ts [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
 
 export default function useIsDebugAddress(address?: string | null) {
-  const [debugAddresses] = useAsyncMemo(() => Governance.get().getDebugAddresses(), [])
-  const isDebugAddress = useMemo(
-    () => !!(address && debugAddresses && debugAddresses.includes(address)),
-    [address, debugAddresses]
-  )
+  const { data: debugAddresses } = useQuery({
+    queryKey: ['debugAddresses'],
+    queryFn: async () => Governance.get().getDebugAddresses(),
```
Comment (andyesp) [inline src/hooks/useIsDebugAddress.ts:10]: `async` here not necessary

File: src/hooks/useProfile.ts [hunk 6]
```diff
@@ -2,9 +2,14 @@ import { useQuery } from '@tanstack/react-query'
...
+  const fetchProfile = async () => {
+    if (!address) return null
+    return profiles.load(address)
+  }
   const { data: profile } = useQuery({
-    queryKey: [`user-${address}-profile`],
-    queryFn: () => (address ? profiles.load(address) : null),
+    queryKey: [`user-${address?.toLowerCase()}-profile`],
+    queryFn: () => fetchProfile(),
+    staleTime: 3.6e6, // 1 hour
```
Comment (andyesp) [inline src/hooks/useProfile.ts:12]: You can move this to the `hooks/constants.ts` file to reuse as `ONE_HOUR_STALE_TIME`

File: src/hooks/useSubscriptions.ts [hunk 7]
```diff
@@ -1,38 +1,43 @@
...
 
 import { Governance } from '../clients/Governance'
 
 export default function useSubscriptions() {
   const [account] = useAuthContext()
-  const [subscriptions, subscriptionsState] = useAsyncMemo(() => Governance.get().getUserSubscriptions(), [account], {
-    callWithTruthyDeps: true,
-    initialValue: [],
+  const queryKey = `subscriptions#${account}`
+  const { data: subscriptions, ...subscriptionsState } = useQuery({
```
Comment (andyesp) [inline src/hooks/useSubscriptions.ts:9]: Can you destructure isLoading and refetch here?

File: src/hooks/useDelegation.ts [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
       if (!address) return EMPTY_DELEGATION
       return await getDelegations(address)
     },
-    [SNAPSHOT_SPACE, address],
-    { initialValue: EMPTY_DELEGATION, callWithTruthyDeps: true }
-  )
+    staleTime: 3.6e6, // 1 hour
+  })
+  return {
+    delegationResult: delegationResult ?? EMPTY_DELEGATION,
```
Comment (andyesp) [inline src/hooks/useDelegation.ts:17]: Can we call this delegation and isDelegationLoading?

File: src/hooks/useCoAuthorsByProposal.ts [hunk 9]
```diff
@@ -1,27 +1,26 @@
...
 
+import useCoauthors from './useCoauthors'
+
 function useCoAuthorsByProposal(proposal: ProposalAttributes | null) {
   const [account] = useAuthContext()
-  const [allCoauthors] = useAsyncMemo(() => Governance.get().getCoAuthorsByProposal(proposal!.id), [proposal], {
-    initialValue: [] as CoauthorAttributes[],
-    callWithTruthyDeps: true,
-  })
+  const allCoauthors = useCoauthors(proposal?.id || '')
```
Comment (andyesp) [inline src/hooks/useCoAuthorsByProposal.ts:12]: See comment in useCoauthors hook

File: src/hooks/useIsProfileValidated.ts [hunk 10]
```diff
@@ -1,23 +1,26 @@
...
-    [address],
-    {
-      initialValue: null,
-    }
-  )
+    staleTime: DEFAULT_QUERY_STALE_TIME,
+  })
 
-  const validationChecked = isProfileValidated !== null
+  const validationChecked = useMemo(() => isBoolean(isProfileValidated), [isProfileValidated])
```
Comment (andyesp) [inline src/hooks/useIsProfileValidated.ts:23]: Do you think it's this memo necessary?

**Review by andyesp — APPROVED** [review]

**Pull request merged**